overlord/devicestate/handlers.go

```diff
@@ -63,13 +70,48 @@ func deviceAPIBaseURL() string {
 }
 
 var (
-	keyLength        = 4096
-	retryInterval    = 60 * time.Second
-	deviceAPIBase    = deviceAPIBaseURL()
-	requestIDURL     = deviceAPIBase + "request-id"
-	serialRequestURL = deviceAPIBase + "devices"
+	keyLength        uint64 = 4096
+	retryInterval           = 60 * time.Second
+	deviceAPIBase           = deviceAPIBaseURL()
+	requestIDURL            = deviceAPIBase + "request-id"
+	serialRequestURL        = deviceAPIBase + "devices"
 )
 
+func rsaGenerateKey() (*rsa.PrivateKey, error) {
```

Comment:
I would call this one generateRSAKey(keyLength int) ...
and at this point probably better to split this out also to a generate_rsa_key.go file